src/drivers/net/mcst/l_e1000.c

```diff
@@ -0,0 +1,728 @@
+	if (ep->base_ioaddr) {
+		e2k_write32(val, ep->base_ioaddr + E_CSR);
+	}
+}
+
+static inline
+uint32_t e1000_read_mgio_csr(struct l_e1000_priv *ep) {
+	if (ep->base_ioaddr) {
+		return e2k_read32(ep->base_ioaddr + MGIO_CSR);
+	}
+	return 0xabcdefab;
+}
+
+static inline
+void e1000_write_mgio_csr(struct l_e1000_priv *ep, int val) {
+	if (ep->base_ioaddr) {
+		e2k_write32(val, ep->base_ioaddr + MGIO_CSR);
+	}
+}
+
+static inline
+uint32_t e1000_read_mgio_data(struct l_e1000_priv *ep) {
+	if (ep->base_ioaddr) {
+		return e2k_read32(ep->base_ioaddr + MGIO_DATA);
+	}
+	return 0xabcdefab;
+}
+
+static inline
+void e1000_write_mgio_data(struct l_e1000_priv *ep, int val) {
+	if (ep->base_ioaddr) {
+		e2k_write32(val, ep->base_ioaddr + MGIO_DATA);
+	}
+}
+
+static inline
+void e1000_write_e_base_address(struct l_e1000_priv *ep, int val) {
+	if (ep->base_ioaddr) {
+		e2k_write32(val, ep->base_ioaddr + E_BASE_ADDR);
+	}
+}
+
+static inline
+void e1000_write_dma_base_address(struct l_e1000_priv *ep, int val) {
+	if (ep->base_ioaddr) {
+		e2k_write32(val, ep->base_ioaddr + DMA_BASE_ADDR);
+	}
+}
+
+static inline
+uint32_t e1000_read_psf_csr(struct l_e1000_priv *ep) {
+	if (ep->base_ioaddr) {
+		return e2k_read32(ep->base_ioaddr + PSF_CSR);
+	}
+	return 0xabcdefab;
+}
+
+static inline
+uint32_t e1000_read_psf_data(struct l_e1000_priv *ep) {
+	if (ep->base_ioaddr) {
+		return e2k_read32(ep->base_ioaddr + PSF_DATA);
+	}
+	return 0xabcdefab;
+}
+
+static int e1000_mii_readreg(struct net_device *dev, int phy_id, int reg_num) {
+	struct l_e1000_priv *ep = netdev_priv(dev);
+	uint32_t rd;
+	uint16_t val_out = 0;
+	int i = 0;
+
+	rd = 0;
+	rd |= 0x2 << MGIO_CS_OFF;
+	rd |= 0x1 << MGIO_ST_OF_F_OFF;
+	rd |= 0x2 << MGIO_OP_CODE_OFF; /* Read */
+	rd |= (phy_id  & 0x1f) << MGIO_PHY_AD_OFF;
+	rd |= (reg_num & 0x1f) << MGIO_REG_AD_OFF;
+
+	e1000_write_mgio_data(ep, rd);
+	rd = 0;
+
+	for (i = 0; i != 1000; i++) {
+		if (e1000_read_mgio_csr(ep) & RRDY) {
+			rd = (uint16_t)e1000_read_mgio_data(ep);
+
+			val_out = rd & 0xffff;
+			log_debug("reg 0x%x >>> 0x%x", reg_num, val_out);
+			return val_out;
+		}
+	}
+
+	log_error("mdio_read: Unable to read from MGIO_DATA reg\n");
+
+	return val_out;
+}
+
+static void e1000_mii_writereg(struct net_device *dev, int phy_id, int reg_num, int val) {
+	struct l_e1000_priv *ep = netdev_priv(dev);
+	uint32_t wr;
+	int i = 0;
+
+	wr = 0;
+	wr |= 0x2 << MGIO_CS_OFF;
+	wr |= 0x1 << MGIO_ST_OF_F_OFF;
+	wr |= 0x1 << MGIO_OP_CODE_OFF; /* Write */
+	wr |= (phy_id  & 0x1f) << MGIO_PHY_AD_OFF;
+	wr |= (reg_num & 0x1f) << MGIO_REG_AD_OFF;
+	wr |= val & 0xffff;
+
+	log_debug("reg 0x%x <<< 0x%x", reg_num, val);
+	e1000_write_mgio_data(ep, wr);
+
+	for (i = 0; i != 1000; i++) {
+		if (e1000_read_mgio_csr(ep) & RRDY) {
+			return;
+		}
+	}
+	log_error("Unable to write MGIO_DATA reg: val = 0x%x", wr);
+	return;
+}
+
+#define SPEED_10   10
+#define SPEED_100  100
+#define SPEED_1000 1000
+static int assigned_speed = SPEED_1000;
+static int half_duplex;
```

Comment:
Why these variables are static and not in priv network card data?

Reply:
I removed both at all